Modules/TPC/src/QualityObserver.cxx

```diff
@@ -0,0 +1,166 @@
+
+  for (const auto& dataSourceConfig : config.get_child("qc.postprocessing." + name + ".qualityObserverConfig")) {
+    Config dataConfig;
+
+    dataConfig.GroupTitle = dataSourceConfig.second.get<std::string>("groupTitle");
+    dataConfig.Path = dataSourceConfig.second.get<std::string>("path");
+
+    std::vector<std::string> inputQO;
+    std::vector<std::string> inputQOTitle;
+
+    if (const auto& QOsources = dataSourceConfig.second.get_child_optional("inputObjects"); QOsources.has_value()) {
+      for (const auto& QOsource : QOsources.value()) {
+        inputQO.push_back(QOsource.second.data());
+      }
+    }
+    if (const auto& QOTitlesources = dataSourceConfig.second.get_child_optional("inputObjectTitles"); QOTitlesources.has_value()) {
+      for (const auto& QOTitlesource : QOTitlesources.value()) {
+        inputQOTitle.push_back(QOTitlesource.second.data());
+      }
+    }
+    if (inputQO.size() != inputQOTitle.size()) {
+      ILOG(Error, Devel) << "in config of group" << dataConfig.GroupTitle << ": Number of QOs does not match number of QO titles!" << ENDM;
+    }
+    dataConfig.QO = inputQO;
+    dataConfig.QOTitle = inputQOTitle;
+    mConfig.push_back(dataConfig);
+
+    inputQO.clear();
+    inputQOTitle.clear();
+  } // for (const auto& dataSourceConfig : config.get_child("qc.postprocessing." + name + ".ratioConfig"))
+}
+
+void QualityObserver::initialize(Trigger, framework::ServiceRegistry&)
+{
+  for (const auto config : mConfig) {
+    Qualities[config.GroupTitle] = std::vector<std::string>();
+  }
+  mColors["bad"] = kRed;
+  mColors["medium"] = kOrange - 3;
+  mColors["good"] = kGreen + 2;
+  mColors["no quality"] = kGray + 2;
+}
+
+void QualityObserver::update(Trigger t, framework::ServiceRegistry& services)
+{
+  auto& qcdb = services.get<repository::DatabaseInterface>();
+  getQualities(t, qcdb);
+  generatePanel();
+}
+
+void QualityObserver::finalize(Trigger t, framework::ServiceRegistry&)
+{
+  generatePanel();
+  if (mCanvas) {
+    delete mCanvas;
+    mCanvas = nullptr;
+  }
+}
+
+void QualityObserver::getQualities(const Trigger& t,
+                                   repository::DatabaseInterface& qcdb)
+{
+  for (const auto& config : mConfig) {
+
+    if (Qualities[config.GroupTitle].size() > 0) {
+      Qualities[config.GroupTitle].clear();
+    }
+    for (const std::string qualityobject : config.QO) {
+      auto qo = qcdb.retrieveQO(config.Path + "/" + qualityobject, t.timestamp, t.activity);
+
+      if (qo) {
+        const auto quality = qo->getQuality();
+        uint qualitylevel = quality.getLevel();
+
+        if (qualitylevel == 1) {
+          Qualities[config.GroupTitle].push_back("good");
+        } else if (qualitylevel == 2) {
+          Qualities[config.GroupTitle].push_back("medium");
+        } else if (qualitylevel == 3) {
+          Qualities[config.GroupTitle].push_back("bad");
+        } else {
+          Qualities[config.GroupTitle].push_back("no quality");
+        }
```

Comment:
Why don't you simply use the official definitions?
https://github.com/AliceO2Group/QualityControl/blob/master/Framework/src/Quality.cxx#L28
```suggestion
   mQualities[config.GroupTitle].push_back(quality.getName());
```

Reply:
Changed as requested